test/samplers/guaranteed_throughput_sampler.js

```diff
@@ -69,53 +69,76 @@ describe('GuaranteedThroughput sampler', () => {
         sampler.close();
     });
 
-    let assertValues = function assertValues(sampler, lb, rate) {
-        assert.equal(lb, sampler._lowerBoundSampler.maxTracesPerSecond);
+    it('should rate limited by upper bound rate limiter', () => {
+        let sampler = new GuaranteedThroughputSampler(1, 1, 2);
+
+        let expectedTags = {'sampler.type': 'probabilistic', 'sampler.param': 1};
+        [1, 2, 3].forEach((_) => {
+            let actualTags = {};
+            assert.isOk(sampler.isSampled('testOperationName', actualTags), 'must sample');
+            assert.deepEqual(expectedTags, actualTags);
+        });
+        // The max_samples_per_second is 2, so after three calls, the upperBoundRateLimiter
+        // should be triggered and the samplingRate should be halved.
+        assertValues(sampler, 0.5, 1, 2);
+
+        sampler.close();
+    });
+
+    let assertValues = function assertValues(sampler, rate, lb, up) {
         assert.equal(rate, sampler._probabilisticSampler.samplingRate);
+        assert.equal(lb, sampler._lowerBoundSampler.maxTracesPerSecond);
+        assert.equal(up, sampler._upperBoundRateLimiter.creditsPerSecond);
     };
 
     it('should not change when update() called with the same values', () => {
-        let sampler = new GuaranteedThroughputSampler(2, 1.0);
-        assertValues(sampler, 2, 1.0);
+        let sampler = new GuaranteedThroughputSampler(1.0, 2, 3);
```

Comment:
When I look at a lot of this code i'm not sure what the parameters mean for GuaranteedThroughputSampler.  Numbers almost always be declared as variables, to convey understanding.  Even better would be defining a GuranteedThroughputSampler in a beforeEach block, since most of them have the same parameters.  If some of the samplers have unique parameters then override them for that test.